test/integration/targets/vmware_guest_disk/tasks/main.yml

```diff
@@ -77,3 +77,156 @@
   assert:
     that:
         - test_create_disk2 is changed
+
+- name: create new disk with custom shares
+  vmware_guest_disk:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: "{{ virtual_machines[0].name }}"
+    disk: 
+        - size_gb: 15
+          type: eagerzeroedthick
+          datastore: "{{ rw_datastore }}"
+          disk_mode: "independent_nonpersistent"
+          scsi_controller: 1
+          state: present
+          unit_number: 4
+          shares:
+            level: custom
+            level_value: 1300
+  register: test_custom_shares
+
+- debug:
+    msg: "{{ test_custom_shares }}"
+
+- name: assert that changes were made
+  assert:
+     that:
+         - test_custom_shares is changed
+ 
+- name: create new disk with custom IO limits and shares in IO Limits
+  vmware_guest_disk:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: "{{ virtual_machines[0].name }}"
+    disk: 
+        - size_gb: 16
+          type: eagerzeroedthick
+          datastore: "{{ rw_datastore }}"
+          disk_mode: "independent_nonpersistent"
+          scsi_controller: 2
+          state: present
+          unit_number: 4
+          iolimit:
+            limit: 1506
+            shares:
+              level: custom
+              level_value: 1305
+  register: test_custom_IoLimit_shares
+
+- debug:
+    msg: "{{ test_custom_IoLimit_shares }}"
+
+- name: assert that changes were made
+  assert:
+     that:
+         - test_custom_IoLimit_shares is changed
+
+- name: Update disk for custom IO limits in IO Limits
+  vmware_guest_disk:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: "{{ virtual_machines[0].name }}"
+    disk: 
+        - size_gb: 17
+          type: eagerzeroedthick
+          datastore: "{{ rw_datastore }}"
+          disk_mode: "independent_nonpersistent"
+          scsi_controller: 2
+          state: present
+          unit_number: 4
+          iolimit:
+            limit: 1500
+            shares:
+              level: custom
+              level_value: 1305
+  register: test_custom_IoLimit
+
+- debug:
+    msg: "{{ test_custom_IoLimit }}"
+
+- name: assert that changes were made
+  assert:
+     that:
+         - test_custom_IoLimit is changed
+
+- name: Update disk for shares of IO limits
+  vmware_guest_disk:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: "{{ virtual_machines[0].name }}"
+    disk: 
+        - size_gb: 18
```

Comment:
The same please.

Reply:
@goneri
as part of integration test we cant maintain 1gb of size in all test cases, as while updating the disk attributes, size should be more than the existing disk size , for now i am giving single digit disk sizes like 1,2,3,4

Reply:
I understand, thanks for the clarification. Could you still reduce the 18GB to something slower?

Reply:
hi @goneri 
thanks for the update, i have already reduced the size to 3 gb , i think you are pointing to the bit older file, you can check following url for the changes what i have made.
url:https://github.com/ansible/ansible/pull/64087/commits/9c53c01e4a19e69a0928bcc480e3550eae9bd0e5